tests/generators/bls/main.py

```diff
@@ -355,7 +358,138 @@ def case05_aggregate_verify():
     }
 
 
-def create_provider(handler_name: str,
+def case06_eth2_aggregate_pubkeys():
+    aggregate_pubkey = spec.eth2_aggregate_pubkeys(PUBKEYS)
+    assert aggregate_pubkey == milagro_bls._AggregatePKs(PUBKEYS)
+    yield f'eth2_aggregate_pubkeys_some_pubkeys', {
+        'input': [encode_hex(pubkey) for pubkey in PUBKEYS],
+        'output': encode_hex(aggregate_pubkey),
+    }
+
+    # Invalid pubkeys -- len(pubkeys) == 0
+    expect_exception(spec.eth2_aggregate_pubkeys, [])
+    expect_exception(milagro_bls._AggregatePKs, [])
+    yield f'eth2_aggregate_pubkeys_empty_list', {
+        'input': [],
+        'output': None,
+    }
+
+    # Invalid pubkeys -- [ZERO_PUBKEY]
+    expect_exception(spec.eth2_aggregate_pubkeys, [ZERO_PUBKEY])
+    expect_exception(milagro_bls._AggregatePKs, [ZERO_PUBKEY])
+    yield f'eth2_aggregate_pubkeys_na_pubkey', {
+        'input': [encode_hex(ZERO_PUBKEY)],
+        'output': None,
+    }
+
+    # Invalid pubkeys -- G1 point at infinity
+    expect_exception(spec.eth2_aggregate_pubkeys, [Z1_PUBKEY])
+    expect_exception(milagro_bls._AggregatePKs, [Z1_PUBKEY])
+    yield f'eth2_aggregate_pubkeys_infinity_pubkey', {
+        'input': [encode_hex(Z1_PUBKEY)],
+        'output': None,
+    }
+
+    # Invalid pubkeys -- b'\x40\x00\x00\x00....\x00' pubkey
+    x40_pubkey = b'\x40' + b'\00' * 47
+    expect_exception(spec.eth2_aggregate_pubkeys, [x40_pubkey])
+    expect_exception(milagro_bls._AggregatePKs, [x40_pubkey])
+    yield f'eth2_aggregate_pubkeys_x40_pubkey', {
+        'input': [encode_hex(x40_pubkey)],
+        'output': None,
+    }
```

Comment:
and perhaps one final random 32-byte value with invalid high bits?

Reply:
ditto